subprojects/jacoco/src/main/java/org/gradle/testing/jacoco/plugins/JacocoTaskExtension.java

```diff
@@ -52,11 +54,12 @@ public String getAsArg() {
         }
     }
 
+    private Project project;
```

Comment:
Two questions about that:

1. I'd assume that we still want to have some sort of method that let's us create a `Provider` from a `Script` for convenience reasons. Correct?
2. I think injecting a factory of a `Provider` into an extension is kind of tricky. We usually create an extension through the extension container. I don't think something like the following would work as the container would want to look for the default constructor - or we'd have to pass in the factory as constructor parameter which requires us to get a hold of it e.g. via `Project`:

```
class MyExtension {
    private final PropertyState<Boolean> enabled
    private final PropertyState<FileCollection> outputFiles

    @javax.inject.Inject
    MyExtension(PropertyStateFactory propertyStateFactory) {
        enabled = propertyStateFactory.property(Boolean)
        outputFiles = propertyStateFactory.property(FileCollection)
    }
}
```